Packet++/header/WireGuardLayer.h

```diff
@@ -0,0 +1,596 @@
+#pragma once
+
+#include "Layer.h"
+#include "IpAddress.h"
+#include "MacAddress.h"
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+
+	/**
+	 * @class WireGuardLayer
+	 * Represents a WireGuard protocol layer
+	 */
+	class WireGuardLayer : public Layer
+	{
+	protected:
+#pragma pack(push, 1)
+		/**
+		 * @struct wg_common_header
+		 * Represents the common header for all WireGuard message types
+		 */
+		struct wg_common_header
+		{
+			/** Message type field */
+			uint8_t messageType;
+			/** Reserved field (3 bytes) */
+			uint8_t reserved[3];
+		};
+#pragma pack(pop)
+		/**
+		 * Get the basic header common to all WireGuard messages.
+		 *
+		 * @return Pointer to the common header structure.
+		 */
+		wg_common_header* getBasicHeader() const
+		{
+			return reinterpret_cast<wg_common_header*>(m_Data);
+		}
+
```

Comment:
nit: private/protected methods or structs don't need doxygen documentation (same in the other classes in this file)

Reply:
Sure, I've made the changes.

Reply:
I added documentation to the protected struct to avoid CI errors.

Reply:
CI won't fail if private/protected methods don't have doxygent documentation, it will only fail if public method don't have it